main/python/cmdLineUtils.py

```diff
@@ -1171,33 +1179,49 @@ def _rootLsPrintSimpleLs(keyList,indent,oneColumn):
         if (i+1)%ncol != 0 and i != len(keyList)-1:
             if not IS_TERMINAL: write( \
                 key.GetName().ljust(col_widths[i%ncol]))
-            elif isDirectoryKey(keyList[i]): write( \
+            elif isDirectoryKey(key):
+                write( \
                 isSpecial(ANSI_BLUE,key.GetName()).ljust( \
                     col_widths[i%ncol] + ANSI_BLUE_LENGTH))
-            elif isTreeKey(keyList[i]): write( \
+                if recursiveListing and len(ROOT.gDirectory.Get(key.GetName()).GetListOfKeys()):
+                    subDir = ROOT.gDirectory.Get(key.GetName())
+                    subkeyList = subDir.GetListOfKeys()
+                    subDir.cd()
+                    write('\n')
+                    _rootLsPrintSimpleLs(subkeyList, indent + 2, oneColumn, recursiveListing)
+                    subDir.cd('../')
+            elif isTreeKey(key): write( \
                 isSpecial(ANSI_GREEN,key.GetName()).ljust( \
                     col_widths[i%ncol] + ANSI_GREEN_LENGTH))
             else: write(key.GetName().ljust(col_widths[i%ncol]))
         else: # No spaces after the last element of the line or of the list
             if not IS_TERMINAL: write(key.GetName())
-            elif isDirectoryKey(keyList[i]):
+            elif isDirectoryKey(key):
                 write(isSpecial(ANSI_BLUE, key.GetName()))
-            elif isTreeKey(keyList[i]):
+                if recursiveListing and len(ROOT.gDirectory.Get(key.GetName()).GetListOfKeys()):
+                    oldDir = ROOT.gDirectory
+                    subDir = ROOT.gDirectory.Get(key.GetName())
+                    subkeyList = subDir.GetListOfKeys()
+                    subDir.cd()
+                    write('\n')
+                    _rootLsPrintSimpleLs(subkeyList, indent + 2, oneColumn, recursiveListing)
+                    subDir.cd('../')
```

Comment:
This few lines are duplicated with what is a few lines before.

Can this code duplication be avoided by putting it into a separate function? In Python you can functions in any scope, so that should easily be possible.